crates/services/sync/src/import.rs

```diff
@@ -364,28 +399,72 @@ fn get_block_stream<
     params: Config,
     p2p: Arc<P>,
     consensus: Arc<C>,
+    cache: SharedMutex<HashMap<Range<u32>, CachedData>>,
 ) -> impl Stream<Item = impl Future<Output = SealedBlockBatch>> {
-    let header_stream = get_header_batch_stream(range.clone(), params, p2p.clone());
-    header_stream
+    let ranges = range_chunks(range, params.header_batch_size);
+    futures::stream::iter(ranges)
+        .map({
```

Comment:
While the pattern was established before this PR, I think it would be nice to use `then` instead of `map` here, and skip the `.awaits`. We'd be able to return just a `Stream<Item = SealedBlockBatch>` instead of having the nested futures in the returned stream.

Reply:
I agree and there is a lot more things to improve on this service I don't wanna make this PR even bigger and so I created an issue for that : https://github.com/FuelLabs/fuel-core/issues/2370